### PR TITLE
Add options to dump T_NONE objects and add page numbers

### PR DESCRIPTION
**Pull request by tenderlove**
This adds options to `dump_all` to dump `T_NONE` slots and add page numbers to each object.  Unfortunately, we can't assign page numbers based on the actual page where the object is stored because `rb_objspace_each_objects` doesn't expose the actual page pointer, though the page number is guaranteed to be correct _per dump_.  IOW, all objects with "page 25" in them are stored on the same page, but if you dump the heap a second time "page 25" may not be the same page as the previous dump.

### CODE REVIEW

File: ext/objspace/objspace_dump.c [hunk 1]
```diff
@@ -347,9 +369,20 @@ dump_output(struct dump_config *dc, VALUE opts, VALUE output, const char *filena
...
 
-    if (RTEST(opts))
+    dc->pages_seen = 0;
+    dc->include_pages = 0;
+    dc->include_none = 0;
+
+    if (RTEST(opts)) {
 	output = rb_hash_aref(opts, sym_output);
 
+	if (Qtrue == rb_hash_lookup2(opts, sym_include_pages, Qfalse))
```
Comment (vmg) [inline ext/objspace/objspace_dump.c]: Are reverse comparisons standard on MRI? I don't recall seeing many of these around.

**Comment by vmg** [comment]
> This adds options to dump_all to dump T_NONE slots and add page numbers to each object. 

This looks great to me. The goal here is monitoring heap fragmentation, right?

File: ext/objspace/objspace_dump.c (hunk 1 above)
Comment (tenderlove) [inline ext/objspace/objspace_dump.c]: No, I think I was just thinking in reverse.  I'll switch it. ;)

**Comment by tenderlove** [comment]
> This looks great to me. The goal here is monitoring heap fragmentation, right?

Correct.  I'll try to send this patch upstream, but I'm not confident that it will be acceptable due to the issues I outlined in the description.

**Pull request closed without merging**